x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsConfigTests.java

```diff
@@ -88,6 +90,49 @@ protected DataFrameAnalyticsConfig createTestInstance() {
         return createRandom(randomValidId(), lenient);
     }
 
+    @Override
+    protected List<Version> bwcVersions() {
+        return Arrays.asList(Version.V_7_0_0,
+            Version.V_7_1_1,
+            Version.V_7_2_1,
+            Version.V_7_3_2,
+            Version.V_7_4_2,
+            Version.V_7_5_3,
+            Version.V_7_6_0,
+            Version.V_7_7_0);
+    }
+
+    @Override
+    protected DataFrameAnalyticsConfig mutateInstanceForVersion(DataFrameAnalyticsConfig instance, Version version) {
+        DataFrameAnalyticsConfig.Builder builder = new DataFrameAnalyticsConfig.Builder(instance);
+        if (version.before(Version.V_7_6_0)) {
+            if (instance.getAnalysis() instanceof Regression) {
+                // TODO regression and classification serialize a randomized value for the random seed
+            }
+            builder.setSource(new DataFrameAnalyticsSource.Builder(instance.getSource())
```

Comment:
This one too. Also, I wonder if it's worth having a `Builder` class for this just for this test. If we had a `DataFrameAnalyticsSourceTests.mutateInstanceForVersion` it could perform this logic just by using the constructor.

Reply:
I can set that up, for sure.